app/controllers/students/tests_controller.rb

```diff
@@ -1,8 +1,17 @@
 class Students::TestsController < ApplicationController
 
+  def index
+    @test_setting = TestSetting.find(params[:test_setting_id])
+    @questions = @test_setting.questions
+    tests_belongs_to_test_setting = Test.where(test_setting_id: @test_setting.id)
+    @attempts_number = tests_belongs_to_test_setting.where(user_id: current_user.id)
+    @test = current_user.tests.last
```

Comment:
if you wanted to show test result, why you redirected to index page? Just make redirect to `show` page, or if it is already used, create new one like `show_result` and route would have next path:
```
some_path_before/:test_id/show_result
```